Spanner/src/Transaction.php

```diff
@@ -615,6 +610,21 @@ public function commit(array $options = [])
             throw new \BadMethodCallException('The transaction cannot be committed because it is not active');
         }
 
+        // For commit, A transaction ID is mandatory for non-single-use transactions,
+        // and the `begin` option is not supported.
+        if (is_null($this->transactionId) && isset($this->options['begin'])) {
+            // Since the begin option is not supported in commit, unset it.
+            unset($this->options['begin']);
```

Comment:
What will happen in case an operation with inline begin fails? Which code flow will ensure the retry happen with a separate begin transaction call?

Reply:
Here it is: https://github.com/googleapis/google-cloud-php/pull/6708/files#diff-0f7d76f7446d5130fcc22c2a4708e82093c2deb6741e9da46f3adc17540e44a8R905